server/src/test/java/org/opensearch/gateway/ReplicaShardBatchAllocatorTests.java

```diff
@@ -692,6 +745,87 @@ private RoutingAllocation onePrimaryOnNode1And1Replica(AllocationDeciders decide
         );
     }
 
+    private RoutingAllocation routingAllocationWithMultipleShards(AllocationDeciders deciders, int primaries, int replicas)
+        throws InterruptedException {
+        return routingAllocationWithMultipleShards(deciders, Settings.EMPTY, primaries, replicas, UnassignedInfo.Reason.CLUSTER_RECOVERED);
+    }
+
+    private RoutingAllocation routingAllocationWithMultipleShards(
```

Comment:
Can we name this method more specific (`routingAllocationWithMultipleDelayAssigned(nUnassigned)`)

Reply:
Would routingAllocationWithMultipleUnassignedShards be a better? this particular method has nothing specific to do with delaying shard assignment,  at high level it is creating shards with their replicas unassigned.